### ISSUE TITLE
DataGrid: More control over edit template

### ISSUE THREAD

**Issue by WolfgangKluge** [Megabit/Blazorise#3700]
I'd like to have a way to manipulate the rendering of the edit form of a DataGrid..

Think of something like a (scholar) dictionary where the words and other parts like genus or numerus should be side by side and are grouped together (if there's not enough space).

The rendering should be something like this:

<table>
  <tr>
    <td colspan="2">Caption 1:<br/>[input field 1]</td>
  </tr>
  <tr>
    <td>Caption 2a:<br/>[input field 2a]</td>
    <td>Caption 2b:<br/>[input field 2b]</td>
  </tr>
  <tr>
    <td>Caption 3a:<br/>[input field 3a]</td>
    <td>Caption 3b:<br/>[input field 3b]</td>
  </tr>
</table>

or on smaller devices:
<table>
  <tr>
    <td colspan="2">Caption 1:<br/>[input field 1]</td>
  </tr>
  <tr>
    <td>Caption 2a:<br/>[input field 2a]</td>
  </tr>
  <tr>
    <td>Caption 3a:<br/>[input field 3a]</td>
  </tr>
  <tr>
    <td>Caption 2b:<br/>[input field 2b]</td>
  </tr>
  <tr>
    <td>Caption 3b:<br/>[input field 3b]</td>
  </tr>
</table>

Other use cases involve colspan (as also shown above) or might want to group some fields together...

So, basically I need a way to customize [_DataGridRowEdit.razor](https://github.com/Megabit/Blazorise/blob/master/Source/Extensions/Blazorise.DataGrid/_DataGridRowEdit.razor).

As a workaround, I use "Edit-Only"-Columns and empty columns so it behaves more or less like I want.

**Comment by stsrki** [comment]
Have you tried `PopupFieldColumnSize` parameter on a DataGridColumn?

**Comment by WolfgangKluge** [comment]
Frankly, since I use the EditMode.Form, I didn't considered that..

This might help, but could not solve all the problems I have (in my example you would render 1, 2a, 2b, 3a, 3b on small devices where I want 1, 2a, 3a, 2b, 3b).

**Comment by WolfgangKluge** [comment]
As expected, `PopupFieldColumnSize` seems to work in Popup-Mode only (but there it's a viable option to control the layout).

**Comment by stsrki** [comment]
Not sure if we would introduce a new `EditFormFieldColumnSize` for Form mode. Or a more generic `EditFieldColumnSize` that would handle both popup and form modes.

**Comment by David-Moreira** [comment]
I am a bit conflicted also... maybe we'll have new edit modes in the future? We never know haha, kinda more In favor of the generic version, and eventually deprecate the previous. Which of course would not do anything in Inline mode.
